controllers/nicclusterpolicy_controller.go

```diff
@@ -182,28 +174,68 @@ func (r *NicClusterPolicyReconciler) updateNodeLabels(
 	return nil
 }
 
-// clearMofedWaitLabel set "network.nvidia.com/operator.mofed.wait" to false
-// on Nodes with Mellanox NICs
+// clearMofedWaitLabel
+// - set "network.nvidia.com/operator.mofed.wait" to false on Nodes with NVIDIA NICs
+// - remove "network.nvidia.com/operator.mofed.wait" which have no NVIDIA NICs anymore
+// - set "network.nvidia.com/operator.mofed.wait" to true if detects OFED Pod
+// on the node (probably in the terminating state)
 func (r *NicClusterPolicyReconciler) clearMofedWaitLabel(ctx context.Context) error {
-	// We deploy OFED and Device plugins only on a nodes with Mellanox NICs
+	nodesWithOFEDContainer := map[string]struct{}{}
+	pods := &corev1.PodList{}
+	if err := r.Client.List(ctx, pods, client.MatchingLabels{consts.OfedDriverLabel: ""}); err != nil {
+		return errors.Wrap(err, "failed to list pods")
+	}
+	for i := range pods.Items {
+		pod := pods.Items[i]
+		if pod.Spec.NodeName != "" {
+			nodesWithOFEDContainer[pod.Spec.NodeName] = struct{}{}
+		}
+	}
 	nodes := &corev1.NodeList{}
-
-	err := r.Client.List(ctx, nodes, client.MatchingLabels{nodeinfo.NodeLabelMlnxNIC: "true"})
-	if err != nil {
+	if err := r.Client.List(ctx, nodes); err != nil {
 		return errors.Wrap(err, "failed to list nodes")
 	}
-
 	for i := range nodes.Items {
-		patch := []byte(fmt.Sprintf(`{"metadata":{"labels":{%q:"false"}}}`, nodeinfo.NodeLabelWaitOFED))
-		err := r.Client.Patch(ctx, &nodes.Items[i], client.RawPatch(types.StrategicMergePatchType, patch))
-		if err != nil {
-			return errors.Wrapf(err, "unable to patch %s node label for node %s",
-				nodeinfo.NodeLabelWaitOFED, nodes.Items[i].Name)
+		node := nodes.Items[i]
+		labelValue := ""
+		if _, hasOFED := nodesWithOFEDContainer[node.Name]; hasOFED {
+			labelValue = "true"
```

Comment:
The name of the function is misleading in this case. Should it maybe be something like `UpdateMofedWaitLabels`?

Reply:
I renamed functions to reduce confusion